app/code/local/Dwyera/Pinpay/Model/PaymentMethod.php

```diff
@@ -271,6 +286,47 @@ protected function _place($payment, $requestType, $request)
     }
 
     /**
+     * Process refund through PinPayment online
+     * @param Varien_Object $payment
+     * @param float $amount
+     * @return $this
+     * @throws Mage_Core_Exception
+     */
+    public function refund(Varien_Object $payment, $amount)
+    {
+        if (!$this->canRefund()) {
+            Mage::throwException(Mage::helper('payment')->__('Refund action is not available.'));
+        }
+
+        if ($payment->getAmountPaid() - $payment->getAmountRefunded() < $amount) {
```

Comment:
be careful of rounding errors here @willtran - worth checking that the difference is not less than 0.5c